### ISSUE TITLE
Monitoring v2 - cannot find CRDs during the installation 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#29171]
**What kind of request is this (question/bug/enhancement/feature request):**
a bug that happens occasionally 

**Steps to reproduce (least amount of steps as possible):**
install monitoring v2 into a downstream cluster.

**Result:**
It errors out saying the required CRDs are missing, but actually, the CRDs are just installed as the first step of the installing processs. 

<img width="1871" alt="Screen Shot 2020-09-25 at 12 02 57 PM" src="https://user-images.githubusercontent.com/6218999/94312549-f65d2000-ff31-11ea-861a-ae2b861c1fa6.png">

**Comment by jiaqiluo** [comment]
This bug happens on rancher:master-head 6319706

- run rancher docker-install 
- add a DO cluster
- go to dashboard -> install monitoring v2 without picking any project 

Results:
<img width="1242" alt="Screen Shot 2020-09-30 at 9 45 36 AM" src="https://user-images.githubusercontent.com/6218999/94714992-bc10cb80-0301-11eb-89e5-7ee9e72e2ebb.png">

```
helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-crd-9.4.200.yaml --version=9.4.200 --wait=true rancher-monitoring-crd /home/shell/helm/rancher-monitoring-crd-9.4.200.tgz
--
Wed, Sep 30 2020 9:38:42 am | Release "rancher-monitoring-crd" does not exist. Installing it now.
Wed, Sep 30 2020 9:38:43 am | creating 4 resource(s)
Wed, Sep 30 2020 9:38:44 am | beginning wait for 4 resources with timeout of 10m0s
Wed, Sep 30 2020 9:38:47 am | creating 1 resource(s)
Wed, Sep 30 2020 9:38:47 am | Watching for changes to Job rancher-monitoring-crd-create with timeout of 10m0s
Wed, Sep 30 2020 9:38:47 am | Add/Modify event for rancher-monitoring-crd-create: MODIFIED
Wed, Sep 30 2020 9:38:47 am | rancher-monitoring-crd-create: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
Wed, Sep 30 2020 9:38:55 am | Add/Modify event for rancher-monitoring-crd-create: MODIFIED
Wed, Sep 30 2020 9:38:55 am | Starting delete for "rancher-monitoring-crd-create" Job
Wed, Sep 30 2020 9:38:56 am | NAME: rancher-monitoring-crd
Wed, Sep 30 2020 9:38:56 am | LAST DEPLOYED: Wed Sep 30 16:38:42 2020
Wed, Sep 30 2020 9:38:56 am | NAMESPACE: cattle-monitoring-system
Wed, Sep 30 2020 9:38:56 am | STATUS: deployed
Wed, Sep 30 2020 9:38:56 am | REVISION: 1
Wed, Sep 30 2020 9:38:56 am | TEST SUITE: None
Wed, Sep 30 2020 9:38:56 am |  
Wed, Sep 30 2020 9:38:56 am | ---------------------------------------------------------------------
Wed, Sep 30 2020 9:38:56 am | SUCCESS: helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-crd-9.4.200.yaml --version=9.4.200 --wait=true rancher-monitoring-crd /home/shell/helm/rancher-monitoring-crd-9.4.200.tgz
Wed, Sep 30 2020 9:38:56 am | ---------------------------------------------------------------------
Wed, Sep 30 2020 9:38:56 am | helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-9.4.200.yaml --version=9.4.200 --wait=true rancher-monitoring /home/shell/helm/rancher-monitoring-9.4.200.tgz
Wed, Sep 30 2020 9:38:57 am | Release "rancher-monitoring" does not exist. Installing it now.
Wed, Sep 30 2020 9:38:58 am | Error: execution error at (rancher-monitoring/templates/validate-install-crd.yaml:16:7): Required CRDs are missing. Please install the rancher-monitoring-crd chart before installing this chart.
```

**Comment by aiyengar2** [comment]
Since this only happens occasionally, maybe Dashboard is encountering a race condition. Afaik, it could either be that
1. The Job that applies the CRDs in `rancher-monitoring-crd` is not being completed before the `helm upgrade` is run for `rancher-monitoring`
2. Although the Job is completed, the CRDs themselves aren't accessible via `kubectl get crd <crd-name>` in the k8s API, which causes the lookup to fail in the `rancher-monitoring` chart.

Currently, we execute this Job as a post-install hook: 
https://github.com/rancher/charts/blob/280468a5002aaeb42c165ffcfac02abe7e746b51/charts/rancher-monitoring/charts-crd/templates/jobs.yaml#L9

Based on the [Helm docs](https://helm.sh/docs/topics/charts_hooks/):
```
10. The library executes the post-install hook (loading hook resources)
11. The library waits until the hook is "Ready"
12. The library returns the release object (and other data) to the client
13. The client exits
```

So unless there's a problem with Helm upstream, the docs do specify that the Job should be completed before `rancher-monitoring-crd`'s install client exits. Since we only install `rancher-monitoring` after that, case 1 is not likely.

Case 2 is difficult to replicate but possible; maybe we need to introduce some sort of delay between installing the CRD chart and installing the actual chart to take care of this?

cc: @ibuildthecloud @dramich since you know more about how the Helm operation pod is deployed onto the cluster

**Comment by mikekuzak** [comment]
Same issue with v2.5.5, terraform 0.13.5, Rancher provider 1.11.0

```
Error: [ERROR] installing App V2: helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-crd-9.4.202.yaml --version=9.4.202 --wait=true rancher-monitoring-crd /home/shell/helm/rancher-monitoring-crd-9.4.202.tgz
Release "rancher-monitoring-crd" does not exist. Installing it now.
creating 5 resource(s)
beginning wait for 5 resources with timeout of 10m0s
creating 1 resource(s)
Watching for changes to Job rancher-monitoring-crd-create with timeout of 10m0s
Add/Modify event for rancher-monitoring-crd-create: MODIFIED
rancher-monitoring-crd-create: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
Add/Modify event for rancher-monitoring-crd-create: MODIFIED
Starting delete for "rancher-monitoring-crd-create" Job
NAME: rancher-monitoring-crd
LAST DEPLOYED: Wed Jan 20 22:19:08 2021
NAMESPACE: cattle-monitoring-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
---------------------------------------------------------------------
SUCCESS: helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-crd-9.4.202.yaml --version=9.4.202 --wait=true rancher-monitoring-crd /home/shell/helm/rancher-monitoring-crd-9.4.202.tgz
---------------------------------------------------------------------
helm upgrade --install=true --namespace=cattle-monitoring-system --timeout=10m0s --values=/home/shell/helm/values-rancher-monitoring-9.4.202.yaml --version=9.4.202 --wait=true rancher-monitoring /home/shell/helm/rancher-monitoring-9.4.202.tgz
Release "rancher-monitoring" does not exist. Installing it now.
Error: execution error at (rancher-monitoring/templates/validate-install-crd.yaml:16:7): Required CRDs are missing. Please install the rancher-monitoring-crd chart before installing this chart.
  on .terraform/modules/k8s_cluster/main.tf line 299, in resource "rancher2_app_v2" "applications":
 299: resource "rancher2_app_v2" "applications" {
ERROR: Job failed: exit status 1
```

**Comment by armsnyder** [comment]
Hi, I wanted to add that this is a problem for those of us relying on automation to set up clusters, particularly when using the [rancher2 Terraform plugin](https://github.com/rancher/terraform-provider-rancher2).

Also we've seen the same issue while installing the "rancher-logging" app.

**Comment by iosifnicolae2** [comment]
I'm facing the same problem, any news?

**Comment by thehejik** [comment]
I'm also seeing this occasionally on windows cluster with `rancher-monitoring-9.4.204-rc04` on rancher:2.5.7.

**Comment by lcotonea** [comment]
I'm facing the same problem too. Note that really first installation was cancelled by a wrong configuration of my storage for graphana. The problem began to occur at my second installation attempt. I've attempted to uninstall the remain CRD, and also remove the targeted namespace, but the problem always occurs.

The **workaround** that I've found:
- Clean all : uninstall the remaining CRD by using the chart uninstaller, then remove the targeted namespace
- Install the monitoring in the previous version (9.4.202 for me today): the according version of CRD will be installed, and you will have the same problem. Don't remove anything (don't try to uninstall the CRD, and don't remove the created namespace).
- Go back into the rancher chart installer, and install the targeted version (9.4.203 for me today): the CRD will be upgraded to the according version, and prometheus will be installed.

**Comment by thehejik** [comment]
Fix for missing v2 monitoring CRD has been successfully validated on `rancher/rancher:master-head 97ca638` with v2 monitoring `14.5.101+up14.5.0` from `dev-v2.6` chart repo `commit: cfd917c824f40d09832595ecddabb4570103aebe` [here](https://github.com/rancher/rancher/issues/32025#issuecomment-857824916)

@aiyengar2 Not sure if I may close this issue, please close it or let me know.

**Comment by mikekuzak** [comment]
@thehejik in which version of Rancher 2.5.x will the fix be available?

**Comment by aiyengar2** [comment]
I think we can close it @thehejik.

@mikekuzak probably Rancher 2.5.10. We are targeting the fix for Rancher 2.6 at the moment and have a backport issue tracking this in https://github.com/rancher/rancher/issues/33015.

**Comment by thehejik** [comment]
Close.
